src/azul/plugins/metadata/hca/transform.py

```diff
@@ -187,12 +191,10 @@ def from_index(self, value: str) -> Optional[JSON]:
             return None
         else:
             value, _, unit = value.partition(' ')
-            try:
-                int(value)
-            except ValueError:
-                assert value and unit, (value, unit)
-            else:
-                unit = None if unit == '' else unit
+            if _ == ' ':
+                u_t = unit if unit != ' ' else None
+                assert value and u_t, (value, unit)
+            unit = None if unit == '' else unit
```

Comment:
This seems a bit obscure. Try a different approach.

Reply:
@hannes-ucsc how about the following to reduce obfuscation?
```python
value, whitespace, unit = value.partition(' ')
if whitespace == ' ':
    reject(unit == ' ', (value, unit), exception=AssertionError)
    assert value and unit, (value, unit) 
if unit == '':
    unit = None
return {'value': value, 'unit': unit}
```

Reply:
`separator` or `sep` would be a better name for that variable.

reject with exception=AssertionError is convoluted

I think the underlying issue is that `partition()` doesn't fail when the separator is absent, forcing you to elaborately detect that case afterwards.

![image](https://user-images.githubusercontent.com/5143256/96808214-4615f680-13cd-11eb-9f1e-c2bea9a6e776.png)

You may need to upgrade more_itertools to get `only`